tfhe/src/integer/client_key/radix.rs

```diff
@@ -131,6 +134,16 @@ impl RadixClientKey {
     pub fn num_blocks(&self) -> usize {
         self.num_blocks
     }
+
+    //Todo Can we remove this?
```

Comment:
why is this needed btw ?

Reply:
We are discussing it in https://github.com/zama-ai/tfhe-rs/pull/1424#discussion_r1710030160.

We cannot convert a `DecompressionKey` to `CudaDecompressionKey` since there is a `ShortintBootstrappingKey` attribute in the fourier domain. The CUDA lib takes BSKs in the standard domain only, so the only way to convert a existing CPU pair of compression keys to the CUDA format is to go from the compressed compression keys. However, I was not able to generate new compressed compression keys from the integer crate without introducing `new_compressed_compression_decompression_keys()` in the integer ClientKey. 

I don't know if we really need that modification and we would like to avoid it if possible.

Reply:
I think that's perfectly fine to have a method on the client key for that

Reply:
Rewriting the question:

Do we want to be able to generate new CUDA compression / decompression keys or are we always converting CPU keys?

Reply:
I think it's better to stick to the same pattern for now, i.e. being able to generate GPU keys from a certain type of keys coming from the CPU

Reply:
unless you disagree/see a reason to generate CUDA keys directly

Reply:
In that case we should be generating new keys. E.g. we cannot convert `ServerKey` to `CudaServerKey`. We can convert ciphertexts but not keys that include BSKs in the Fourier domain, as `ServerKey`, `DecompressionKey`, etc. Conversion from a compressed key is possible but not the main use case.

I will implement a method for that.

Reply:
Done.

Now `integer::ClientKey` has a new `new_cuda_compression_decompression_keys()` method. However, to generate a new packing ks key I need access to `RadixClientKey::ClientKey`, which was not public. I changed the visibility to  `pub(crate)`. Wdyt?

Reply:
no need you should be able to do radix_client_key.as_ref() and get a &ClientKey I think